src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopyArray = [];
+  const stateCopy = Object.assign({}, state);
+
+  for (const i of actions) {
+    switch (i.type) {
+      case 'addProperties': Object.assign(stateCopy, i.extraData);
+        break;
+
+      case 'removeProperties': i.keysToRemove.forEach((el) => {
+        delete stateCopy[el];
+      }, {});
+        break;
+
+      case 'clear': Object.keys(stateCopy).forEach((key) => {
+        delete stateCopy[key];
```

Comment:
good but can be more simplify, you can just set empty object as value of `stateCopy`